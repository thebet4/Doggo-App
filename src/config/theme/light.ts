const light = {
  colors: {
    primary: {
      100: "#E6F3FD",
      200: "#BCE0FA",
      300: "#51B1FA",
      400: "#017EE1",
      500: "#015FA9",
      600: "#01487F",
      700: "#013055",
      800: "#012440",
      900: "#011B30",
    },
    success: {
      50: "#E8F0E4",
      100: "#D4E3CB",
      200: "#C0D6B2",
      300: "#82AE67",
      400: "#45861D",
      500: "#307904",
      600: "#2D6B08",
      700: "#2A5D0C",
      800: "#27500E",
      900: "#234410",
    },
    warning: {
      50: "#FEF2E9",
      100: "#FEE7D6",
      200: "#FDCFB0",
      300: "#FCB889",
      400: "#FA8A3C",
      500: "#F97316",
      600: "#DC6B1D",
      700: "#C06322",
      800: "#A65A26",
      900: "#8E5228",
    },
    error: {
      50: "#F8E7EA",
      100: "#F1D2D8",
      200: "#E4A8B3",
      300: "#D1697C",
      400: "#B71532",
      500: "#B00020",
      600: "#9B0722",
      700: "#870C23",
      800: "#741123",
      900: "#521621",
    },
    gray: {
      100: "#FFFFFF",
      200: "#E9ECEF",
      300: "#DEE2E6",
      400: "#CED4DA",
      500: "#ADB5BD",
      600: "#6C757D",
      700: "#495057",
      800: "#343A40",
      900: "#212529",
    },
  },
  fontConfig: {
    Inter: {
      100: {
        normal: "Inter_100Thin",
      },
      200: {
        normal: "Inter_200ExtraLight",
      },
      300: {
        normal: "Inter_300Light",
      },
      400: {
        normal: "Inter_400Regular",
      },
      500: {
        normal: "Inter_500Medium",
      },
      600: {
        normal: "Inter_600SemiBold",
      },
      700: {
        normal: "Inter_700Bold",
      },
      800: {
        normal: "Inter_800ExtraBold",
      },
      900: {
        normal: "Inter_900Black",
      },
    },
  },
};

export default light;
