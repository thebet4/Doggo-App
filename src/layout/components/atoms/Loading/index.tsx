import React from "react";
import { Spinner } from "./styles";

interface ILoadingProps {
  size?: "small" | "large";
}

const Loading = (
  { size }: ILoadingProps = {
    size: "large",
  }
) => {
  return <Spinner size={size} />;
};

export default Loading;
