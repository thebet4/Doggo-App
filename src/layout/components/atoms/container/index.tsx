import React, { ReactElement } from "react";
import { SafeArea } from "./styles";
import { StatusBar } from "expo-status-bar";

type ContainerProps = {
  children: ReactElement;
};

const Container = ({ children }: ContainerProps): JSX.Element => {
  return (
    <SafeArea>
      <StatusBar style="dark" />
      {children}
    </SafeArea>
  );
};
export default Container;
