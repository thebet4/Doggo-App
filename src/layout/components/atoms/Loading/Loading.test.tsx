import React from "react";
import renderer from "react-test-renderer";

import { Loading } from "@layout/components/atoms";
import { ThemeProvider } from "styled-components/native";
import light from "@config/theme/light";

it(`renders <Loading /> correctly`, () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={light}>
        <Loading />
      </ThemeProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
