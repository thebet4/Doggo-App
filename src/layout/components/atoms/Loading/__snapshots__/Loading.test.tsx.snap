// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Loading /> correctly 1`] = `
<ActivityIndicator
  color="#011B30"
  style={
    Array [
      Object {},
    ]
  }
/>
`;
